packages/blockchain-link/README.md

```diff
@@ -1,55 +1,83 @@
-Currently supported:
+blockchain-link is a client and unified interface for several backends (_BE_ further on) of various blockchain networks. Currently, there are implementations for
+- [blockbook](https://github.com/trezor/blockbook): BE developed and deployed by SatoshiLabs. Provides access to Bitcoin(like) and Ethereum(like) networks.
+- [ripple](https://xrpl.org/): third party BE that provides access to the Ripple network.
+
+## API
+
+The API aims to be identical across all backends and networks. However, this isn't entirely possible and there are some discrepancies which are noted below.
+
+All methods can fail and return an error eg. in case the BE is offline.
+
+- `getInfo`: Get general information about the network, like current block-height, name, smallest possible division, etc.
+- `getBlockHash`: Get hash of a block of given height.
+- `getAccountInfo`: Get info about an account, eg. derived addresses, balance, transaction history etc.
+- `getAccountUtxo`: Get unspent inputs for given account. Only bitcoin(like) networks.
+- `getAccountBalanceHistory`: Get historical progression of given account's balance. Used for rendering a graph in Suite's dashboard.
+- `getCurrentFiatRates`: Get current fiat rates.
+- `getFiatRatesForTimestamps`: Get historical fiat rates, only some networks support this.
+- `getFiatRatesTickersList`: Get fiat currencies for which rates are available.
+- `estimateFee`: Get ‘recommended’ fee value inferred from current traffic in the network.
+- `subscribe`: Subscribe for live changes in
+    - blockchain i.e new blocks mined.
+    - accounts, addresses i.e. new transactions broadcasted or mined.
+    - fiatRates
+    - connection to BE (HANDSHAKE, CONNECT, DISCONNECT)
+    
+    Handling subscription state is left to the user. (Hence the purpose of HANDSHAKE, CONNECT and DISCONNECT notifications.)
+- `unsubscribe`: Cancel existing subscription.
+- `getTransaction`: Get info about a given transaction. Return value of this method is not identical across networks; this method exposes the specific transaction format.
+- `pushTransaction`: Broadcast given transaction to the network.
+
+## Usage
+
+Add blockchain-link to your dependencies.
+
+```shell
+yarn add @trezor/blockchain-link
+```
 
--   [blockbook](https://github.com/trezor/blockbook/)
--   [ripple-lib](https://github.com/ripple/ripple-lib/)
+And use it.
 
-## Development:
+```javascript
+import BlockchainLink from '@trezor/blockchain-link';
 
-Compile and run with simple ui for the API methods.
+const link = new BlockchainLink({
+    name: 'Name used in logs.';
+    worker: 'path/to/the/worker.js';
+    server: ['url1.of.the.be', 'url2.of.the.be'];
+    debug: true;
+});
 
+try {
+    const resp = link.getInfo();
+} catch(error) {
+
+}
 ```
+
+## Development
+
+This package provides a simple testing UI for playing around with various implementations and BEs. Run it with
+
+```shell
 yarn
 yarn dev
 ```
 
-## Build
+### Build
```

Comment:
It would be nice to also have a section on publishing. @szymonlesisz could you elaborate on that?

Reply:
sure, you can put this in to separate file `publishing.md` and do a link from here
```
## Publishing `@trezor/blockchain-link` library
__Make sure that you do have npm account and access to write to @trezor/blockchain-link package.__

1. Make sure that CHANGELOG is up to date, usually it is not so walk thru commits and update it.
1. Manually change version number in `./packages/blockchain-link/package.json` file

### Production
1. `cd ./packages/blockchain-link` - go to the root of package
1. `yarn build:lib` - build library
1. `npm publish` - publish library

### Beta

If you want to publish to npm as `beta` (from any branch) do the following:

1. `cd ./packages/blockchain-link` - go to the root of package
1. `vim ./package.json`, manually change version from `X.X.X` to `X.X.(X + 1)-beta.1` [2]
1. `yarn build:lib` - build library
1. `npm publish --tag beta` - publish library

[2] Versioning in npm is sensitive and cannot be reverted. Once you publish a version it is "gone" forever.
For example if the current version is `1.0.17` and you publish `1.0.18` by mistake with custom `beta` tag the next official version will have to be `1.0.19`
This is why the suffix `1.0.17-beta.[number]` in published package.json is important. You can do another releases with `-beta.2` `-beta.3` etc. instead of bumping real version patch number.

```

Reply:
Added (slightly edited) in de4db0e8b6fa2f572e587b52a45142c7bd1e9824